schemas/dns.yml

```diff
@@ -0,0 +1,210 @@
+
+        This is useful to detect data exfiltration via DNS queries.
+      example: 42
+
+    - name: answers_count
+      level: extended
+      type: long
+      description: >
+        The number of resource records contained in the `dns.answers` field.
+      example: 3
+
+    - name: answers
+      level: extended
+      type: object
+      short: Array of DNS answers.
+      description: >
+        An array containing an object for each answer section returned by
+        the server.
+
+        The main keys that should be present in these objects are defined by ECS.
+        Records that have more information may contain more keys than what
+        ECS defines.
+
+        Not all DNS data sources give all details about DNS answers.
+        At minimum, answer objects must contain the `data` key.
+        If more information is available, map as much of it to ECS as possible,
+        and add any additional fields to the answer objects as custom fields.
+
+    - name: answers.name
+      level: extended
+      type: keyword
+      short: The domain name to which this resource record pertains.
+      description: >
+        The domain name to which this resource record pertains.
+
+        If a chain of CNAME is being resolved, each answer's `name` should be
+        the one that corresponds with the answer's `data`. It should not simply
+        be the original `question.name` repeated.
+      example: www.google.com
+
+    - name: answers.type
+      level: extended
+      type: keyword
+      description: The type of data contained in this resource record.
+      example: CNAME
+
+    - name: answers.class
+      level: extended
+      type: keyword
+      description: The class of DNS data contained in this resource record.
+      example: IN
+
+    - name: answers.ttl
+      level: extended
+      type: long
+      description: >
+        The time interval in seconds that this resource record may be cached
+        before it should be discarded. Zero values mean that the data should
+        not be cached.
+      example: 180
+
+    - name: answers.data
+      level: extended
+      type: keyword
+      short: The data describing the resource.
+      description: >
+        The data describing the resource.
+
+        The meaning of this data depends on the type and class of the resource
+        record.
+      example: 10.10.10.10
+
+    - name: grouped.ip
+      level: extended
+      type: ip
+      short: Array containing all IPs seen in answers.data
+      description: >
+        Array containing all IPs seen in `answers.data`.
+
+        The `answers` array can be difficult to use, because of the variety of
+        data formats it can contain. Extracting all IP addresses seen in there to
+        `dns.grouped.ip` makes it possible to index them as IP addresses, and
+        makes them easier to visualize and query for.
+      example: [10.10.10.10, 10.10.10.11]
+
+    - name: grouped.domain
```

Comment:
I'm questioning whether or not this field is necessary. For the search use-case you could query `dns.answers.data` to find a document based a domain. What other uses would there be?

Reply:
Ok let's take it out for now and add it in if we see a concrete use later.

I was under the impression that we could easily format a field that contains an array as one element per line in Kibana index patterns. Turns out there's no obvious way to do it. No support in field formatting, and doing a String join in a scripted field is escaped, whether it's `\n` or HTML.

So the array works for IPs for display because their size is pretty regular (at least for IPv4), so it's pretty readable. But not so for domains.